lib/connextclient/ConnextClient.ts

```diff
@@ -157,17 +157,22 @@ class ConnextClient extends SwapClient {
         tokenAddress: expectedTokenAddress,
       } = expectedIncomingTransfer;
     const currency = this.getCurrencyByTokenaddress(tokenAddress);
-    if (
+    if (rHash === expectedHash) {
+      // Ignore displaying the error if the hash matches, but
+      // tokenAddress, units or timelock is different.
+      // This can happen when Connext is used for both maker
+      // and taker currency.
+      if (
+        tokenAddress === expectedTokenAddress &&
         units === expectedUnits &&
-        rHash === expectedHash &&
-        timelock === expectedTimelock &&
-        tokenAddress === expectedTokenAddress
+        timelock === expectedTimelock
       ) {
-      this.logger.debug(`accepting incoming transfer with rHash: ${rHash}, units: ${units}, timelock ${timelock} and currency ${currency}`);
-      this.emit('htlcAccepted', rHash, units, currency);
-      this.expectedIncomingTransfers.delete(rHash);
+        this.logger.debug(`accepting incoming transfer with rHash: ${rHash}, units: ${units}, timelock ${timelock} and currency ${currency}`);
+        this.emit('htlcAccepted', rHash, units, currency);
+        this.expectedIncomingTransfers.delete(rHash);
+      }
     } else {
-      this.logger.error(`ignoring received pending transfer because it does not meet the requirements -
+      this.logger.warn(`ignoring received pending transfer because it does not meet the requirements -
```

Comment:
We're really only going to be logging this if `rHash !== expectedHash` so I don't think the wording of "does not meet the requirements" applies. It also seems like we won't ever be reaching this code, since if the hashes do not match then above where we do `expectedIncomingTransfers.get(rHash)` we'll get back `undefined` and return from the method. So maybe we should return the above to a warn saying we don't recognize the has, and then here we can assume `rHash === expectedHash` without explicitly checking for it as we're doing above.

Regarding the `CREATED` and `UNLOCKED` events, is there a way we can check which event we're dealing with and log accordingly? Basically if it's an expected condition, I don't even think we'd need to log a trace line. But if we're getting requests for unexpected hashes, a warning is probaly more appropriate.

Reply:
> if the hashes do not match then above where we do expectedIncomingTransfers.get(rHash) we'll get back undefined and return from the method.

You're right. I ended up removing the `"does not meet the requirements"` conditional and the reduntant `rHash === expectedHash` check.

> Regarding the CREATED and UNLOCKED events, is there a way we can check which event we're dealing with and log accordingly?

The `onTransferReceived` function only deals with `CREATED` events. I've clarified the warn logging accordingly.